File: SearchInsertPosition.java
public class Solution {
	public int searchInsert(int[] A, int target) {
		for(int i = 0; i < A.length; i++) {
			if(A[i] >= target) {
				return i > 0 ? i : 0;
			}
		}
		return A.length;
	}
}

File: SearchInsertPosition.py
class Solution:
    # @param A, a list of integers
    # @param target, an integer to be inserted
    # @return integer
    def searchInsert(self, A, target):
        return self.binarySearchInsert(A, 0, len(A), target)
    
    def binarySearchInsert(self, A, begin, end, target):
        if begin >= end:
            return end
        mid = begin + (end - begin) / 2
        if A[mid] == target:
            return mid
        if A[mid] > target:
            return self.binarySearchInsert(A, begin, mid, target)
        return self.binarySearchInsert(A, mid + 1, end, target)